File: MagicalPet.cpp
#include "MagicalPet.h"
#include <iostream>

MagicalPet::MagicalPet(std::string name, std::string type, int age, std::string magicType)
    : VirtualPet(name, type, age), magicType(magicType) {
}

void MagicalPet::showStatus() const {
    
    VirtualPet::showStatus();
    std::cout << "[Магия: " << magicType << "]\n";
}

void MagicalPet::specialAbility() {
    
    std::cout << name << " сотворил " << magicType << "-иллюзию!\n";
}

std::unique_ptr<PetBase> MagicalPet::clone() const {
    auto copy = std::make_unique<MagicalPet>(name, type, age, magicType);
    copy->getParameters() = getParameters(); // копируем параметры
    return copy;
}

File: MagicalPet.cs
using System;

public class MagicalPet : VirtualPet
{
    protected string magicType;

    public MagicalPet(string name, string type, int age, string magicType)
        : base(name, type, age)
    {
        this.magicType = magicType;
    }

    public string GetMagicType() => magicType;

    public override void ShowStatus()
    {
        base.ShowStatus();
        Console.WriteLine("╔════════════════════════════════════════════════════════╗");
        Console.WriteLine($"║  Магический тип: {magicType,-35}   ");
        Console.WriteLine("╚════════════════════════════════════════════════════════╝");
    }
}